tapfreighter/wallet.go

```diff
@@ -385,23 +394,34 @@ func (f *AssetWallet) FundPacket(ctx context.Context,
 		AssetID:  &fundDesc.ID,
 		MinAmt:   1,
 	}
-	eligibleCommitments, err := f.cfg.CoinSelector.ListEligibleCoins(
-		ctx, constraints,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to complete coin selection: %w",
-			err)
-	}
 
-	log.Infof("Identified %v eligible asset inputs for send of %d to %x",
-		len(eligibleCommitments), fundDesc.Amount, fundDesc.ID[:])
+	var selectedCommitments []*AnchoredCommitment
+	err := f.cfg.CoinSelector.WithCoinLock(func() error {
+		eligible, err := f.cfg.CoinSelector.ListEligibleCoins(
+			ctx, constraints,
+		)
+		if err != nil {
+			return fmt.Errorf("unable to complete coin selection: "+
+				"%w", err)
+		}
 
-	selectedCommitments, err := f.cfg.CoinSelector.SelectForAmount(
-		fundDesc.Amount, eligibleCommitments, PreferMaxAmount,
-	)
+		log.Infof("Identified %v eligible asset inputs for send of %d "+
+			"to %x", len(eligible), fundDesc.Amount, fundDesc.ID[:])
+
+		selectedCommitments, err = f.cfg.CoinSelector.SelectForAmount(
```

Comment:
Looks like leasing the coins is missing here? Otherwise two callers can come in series, then obtain the same set of coins, and attempt to lease them both.

Reply:
Yes, that's added in a later commit. Added a TODO here for now to help with the review flow.